gsoc/2021.md

```diff
@@ -0,0 +1,505 @@
+|Prerequisites        | Knowledge of Scala as well as front-end development|
+|Estimated difficulty | Medium|
+|Potential mentors    | [Krzysztof Romanowski], [Mateusz Ziarko]|
+
+### Integrating Metals with bazel-bsp
+
+|Title                | Integrating Metals with bazel-bsp |
+|---------------------|-|
+|Brief description    | Implement support for [Bazel](https://bazel.build/) in [Metals] using [bazel-bsp](https://github.com/JetBrains/bazel-bsp) to take advantage of JetBrains’s efforts to have a build server protocol integration (BSP) for Bazel. This would entail adding support to globs to BSP, ironing out integration quirks and benchmarking performance. The Bazel IJ plugin is often many versions behind the current Bazel version. This effort would enable us to support new versions quickly for all tools that require Bazel integration such as metals.|
+|Expected outcome     | A working proof of concept which uses Metals with bazel-bsp and a roadmap for how to productionize the support.|
+|Prerequisites        | Familiarity with Bazel, LSP, BSP, and VSCode.|
+|Estimated difficulty | Medium |
+|Potential mentors    | [Felix Mulder]|
+
+
+### Creative Coding in Scala
+|Title                | Creative Coding in Scala |
+|---------------------|-|
+|Brief description    | Creative coding is the practice of writing code for some artistic result. Scala has a creative coding 2D graphics library called Doodle. Doodle is built squarely on functional programming techniques, which makes it quite different from better known creative coding environments such as Processing. See [the website](https://creativescala.com/doodle/index.html) for an overview of Doodle and its design principles. The goal of this project is to extend Doodle to allow more creative coding possibilities. There are many enhancements that would both benefit Doodle and form a well-defined project. Examples include: 1. Adding support for bitmap graphics. The most basic form of this already exists, but it could do with many enhancements to allow, for example, directly drawing bitmaps or performing convolution filters on bitmaps. 2. Adding basic support for data visualisation. Building a full data visualisation library is probably too big a task for GSOC but building some basic functionality would be attainable. 3. Backend specific features, such as converting fonts to paths in the Java2D backend, or adding filters in the SVG backend. |
+|Expected outcome     | A complete implementation of the chosen project, including code, tests, and documentation. |
+|Prerequisites        | Required: intermediate Scala knowledge, interest in creative coding or data visualization, interest in functional programming idioms. Preferred: some background knowledge of graphics toolkits will be useful for most projects |
+|Estimated difficulty | Medium |
+|Potential mentors    | [Felix Mulder]|
+
+
+Title: Creative Coding in Scala

```

Comment:
There are two projects here that don't follow the template used by the other ones. Probably an oversight?

Reply:
> We should probably add a link to where the students can apply (unless we don't have it yet?).

I think that they apply through the gsoc portal and I do not know if we have access to the application link